opentasks/src/main/java/org/dmfs/tasks/widget/SubtaskView.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.tasks.widget;
+
+import android.content.Context;
+import android.databinding.DataBindingUtil;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.FrameLayout;
+
+import org.dmfs.android.contentpal.RowDataSnapshot;
+import org.dmfs.opentaskspal.readdata.DueTime;
+import org.dmfs.opentaskspal.readdata.PercentComplete;
+import org.dmfs.opentaskspal.readdata.TaskColor;
+import org.dmfs.opentaskspal.readdata.TaskTitle;
+import org.dmfs.optional.Optional;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.tasks.R;
+import org.dmfs.tasks.contract.TaskContract.Tasks;
+import org.dmfs.tasks.databinding.OpentasksViewItemTaskDetailsSubtaskBinding;
+import org.dmfs.tasks.utils.DateFormatter;
+import org.dmfs.tasks.utils.DateFormatter.DateFormatContext;
+import org.dmfs.tasks.utils.SubtaskTaskDetailsUi;
+
+
+/**
+ * {@link View} for showing a subtask on the details screen.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class SubtaskView extends FrameLayout implements SmartView<RowDataSnapshot<Tasks>>
+{
+
+    /**
+     * The projection required to be able to display this view.
+     */
+    public static final String[] TASK_PROJECTION = new String[] {
+            Tasks._ID,
+            Tasks.TITLE,
+            Tasks.TASK_COLOR,
+            Tasks.LIST_COLOR,
+            Tasks.DUE
+    };
```

Comment:
This is why the progress bar doesn't appear in the details view, you just don't load it (so it's always returned as `null`). You also have to make sure to load `TIMEZONE` and `IS_ALLDAY` in order to get `DUE` right. That might explain the "Jan 1st 1970" issue, not sure.

Reply:
Right, I've added them.

What do you think about instead of passing `RowDataSnapshot`s directly to UI, using specific data/model types that the actual UI needs?
In general this may require creating many of these special, 'UI-local' data types but it decouples the UI from the internals of the data layer representations.

Something like this:
```java
interface SubtasksUiData extends Iterable<SubtaskUiData>
{
      Color parentTaskColor();
}

interface SubtaskUiData
{
       CharSequence title();
       Optional<DateTime> due();
       Optional<Integer> percentComplete();
}
```
So `SubtatskView` could be `SmartView<SubtasksUiData>`,
`SubtaskUiData` would of course adapt `RowDataSnapshot` using the smaller adapters. There could be less potential for this kind of problem of missing columns from the projection.
(Maybe the small adapters could also expose the required projection in a constant? Not very elegant.. not sure what's the best way to handle that clearly.)